### PR TITLE
Fix some issues with blocking and edge generation 

### PR DESCRIPTION
**Pull request by gnieto**
I've found some edge cases where the control flow graph was not properly generated and it was disconnected, which should not be possible on properly formed codes. I've added some test cases for each one of the cases.

The generated graph was disconnected under the following circumstances:

- When an unconditional jump was found, an edge to the following block was added, but the edge target should be to the target address of the jump
- As the final instruction was not checked, if the final instruction was an unconditional jump, the edge was not added.
- As the final instruction was not checked, if the final instruction was a conditional jump, the edge to the jump address was not added
- If the final instruction of a block is a non-branching instruction, an edge to the following block has to be added.

### CODE REVIEW

File: src/cfg.rs [hunk 1]
```diff
@@ -194,4 +246,154 @@ mod tests {
         let outbound = cfg.graph.externals(EdgeDirection::Outgoing);
         assert_eq!(outbound.count(), 1);
     }
+
+    #[test]
+    fn build_cfg_containing_local_unconditional_jump() {
```
Comment (gnieto) [inline src/cfg.rs]: I don't want to be pedantic, but I'm attaching images of how it was formatted before this PR and another one of the CFG on this branch. I think it helps to understand the cases I've changed:
**old**
dot_local_unc_jump
**new**
new_unc_jump

File: src/cfg.rs [hunk 2]
```diff
@@ -194,4 +246,154 @@ mod tests {
...
+        let neighbours = cfg.graph.neighbors(falltrough_idx).into_iter().collect();
+        assert_neighbours(neighbours, vec![ret_idx]);
+
+        let neighbours = cfg.graph.neighbors(ret_idx).into_iter().collect();
+        assert_neighbours(neighbours, vec![]);
+    }
+
+    #[test]
+    fn build_cfg_with_goto_ending() {
+        let insts = [
```
Comment (gnieto) [inline src/cfg.rs]: **old**
dot_ending_goto
**new**
new_goto_ending

File: src/cfg.rs [hunk 3]
```diff
@@ -194,4 +246,154 @@ mod tests {
...
+
+        let neighbours = cfg.graph.neighbors(ret_idx).into_iter().collect();
+        assert_neighbours(neighbours, vec![]);
+
+        let neighbours = cfg.graph.neighbors(goto_idx).into_iter().collect();
+        assert_neighbours(neighbours, vec![ret_idx]);
+    }
+
+    #[test]
+    fn build_cfg_with_branch_ending() {
```
Comment (gnieto) [inline src/cfg.rs]: **old**
dot_branch_ending
**new**
new_branch_ending

File: src/cfg.rs (hunk 1 above)
Comment (waywardmonkeys) [inline src/cfg.rs]: I love the graphs, thanks. :) They help make things easier to see.

**Comment by waywardmonkeys** [comment]
Thank you again!

**Comment by emoon** [comment]
Cool stuff! :)
